### PR TITLE
single '-' or '.' are now parsed as characters rather than numeric

### LINKED ISSUE #297: read_delim converts "." to "0"
The problem is that "." is parsed as a number like ".0", and as a result it returns "0".  That's why adding locale = locale(decimal_mark = ",", grouping_mark = "_") to read_delim helps. Here is an example:

```
> file = "chr1\t1\t10\t.\t15\t20\nchr1\t5\t15\t.\t20\t25"
> read_delim(file, delim="\t", col_names=F)
    X1 X2 X3 X4 X5 X6
1 chr1  1 10  0 15 20
2 chr1  5 15  0 20 25
```

Apart of that it converts "+" and "-" to NA or "0" and that I just do not understand, e.g:

```
> file = "chr1\t1\t10\t+\t15\t20\nchr1\t5\t15\t-\t20\t25"
> read_delim(file, delim="\t", col_names=F)
Warning: 1 parsing failure.
row col expected actual
  1  X4 a number       
    X1 X2 X3 X4 X5 X6
1 chr1  1 10 NA 15 20
2 chr1  5 15  0 20 25
```

### PR DESCRIPTION
**Pull request by jimhester**
Explicitly checking for this case seems like the best option. I played around with editing the DFA in parseNumber but it ended up being more hassle than it was worth.

Fixes #297

### CODE REVIEW

File: src/CollectorGuess.cpp [hunk 1]
```diff
@@ -50,6 +50,10 @@ bool isNumber(const std::string& x, LocaleInfo* pLocale) {
   if (x[0] == '0' && x.size() > 1 && x[1] != pLocale->decimalMark_)
     return false;
 
+  // Only a - or decimal mark
+  if (x.size() == 1 && x[0] == '-' || x[0] == pLocale->decimalMark_)
+    return false;
+
```
Comment (hadley) [inline src/CollectorGuess.cpp]: I think the problem with only fixing it here is that then `parse_numeric(c(".". "-"))` might give incorrect values.

**Comment by hadley** [comment]
I think the problem with the DFA is that nothing checks that the DFA actually processes any characters - i.e. we need some way to distinguish a `sum` set to 0 from processing a zero, and `sum` that remains at zero. 

Can you try adding an additional `bool seenNumber` that's set to true in any of the `*cur >= '0' && *cur <= '9'` blocks? Then instead of `return true` you'd `return seenNumber`.

**Comment by jimhester** [comment]
Ok did it that way with https://github.com/hadley/readr/pull/418/commits/73d146c7aecc0a395795d468f6381ab8ede9bf00 and added a test for the `parse_numeric()` case.

**Comment by hadley** [comment]
Thanks!

**Pull request merged**